File: training-data/swift/acronym.swift
func abbreviate(_ phrase: String) -> String {
    phrase
        .split { !$0.isLetter && $0 != "'" }
        .compactMap { $0.first?.uppercased() }
        .joined()
}

print(abbreviate("Portable Network Graphics"))
print(abbreviate("Complementary metal-oxide semiconductor"))

File: training-data/rust/acronym.rs
fn abbreviate(phrase: &str) -> String {
    phrase
        .split(|c: char| c.is_whitespace() || c == '-' || c == '_')
        .filter_map(|w| w.chars().find(|c| c.is_alphabetic()))
        .map(|c| c.to_ascii_uppercase())
        .collect()
}

fn main() {
    println!("{}", abbreviate("Portable Network Graphics"));
    println!("{}", abbreviate("Complementary metal-oxide semiconductor"));
}
